pkg/apimachinery/patch/patch_test.go

```diff
@@ -75,4 +76,48 @@ var _ = Describe("PatchSet", func() {
 		patches := patch.New()
 		Expect(patches.IsEmpty()).To(BeTrue())
 	})
+
+	DescribeTable("should add the patch bytes", func(p string, expectErr bool) {
+		err := patch.New().AddPatch([]byte(p))
+		if expectErr {
+			Expect(err).To(HaveOccurred())
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+	},
+		Entry("with a wrong patch, it should generate an error", `{"something": "wrong"}`, true),
+		Entry("with a valid patch", `[{"op":"replace","path":"/abcd","value":"test"}]`, false),
+	)
+
+	DescribeTable("should unmarshal the patch", func(operation *patch.PatchOp, expected string) {
+		var t string
```

Comment:
Is there a case where `t` is not a string? `patchSet.Unmarshal` accepts and general `interface{}`, so I am trying to understand why.

Reply:
It could be any object, but in this test the value is a string.